Scripts/DowloadPortalItemData.swift

```diff
@@ -241,39 +231,56 @@ let previousDownloadedItems: DownloadedItems = {
 }()
 var downloadedItems = previousDownloadedItems
 
-// Asynchronously downloads portal items.
-let dispatchGroup = DispatchGroup()
-
-portalItems.forEach { portalItem in
-    let destinationURL = downloadDirectoryURL.appendingPathComponent(portalItem.identifier, isDirectory: true)
-    // Checks if a directory exists or not, to see if an item is already downloaded.
-    if FileManager.default.fileExists(atPath: destinationURL.path) {
-        print("info: Item \(portalItem.identifier) has already been downloaded.")
-    } else {
+await withTaskGroup(of: Void.self) { group in
+    for portalItem in portalItems {
+        // Checks to see if an item is already downloaded.
+        guard downloadedItems[portalItem.identifier] == nil else {
```

Comment:
```suggestion
        guard !downloadedItems.keys.contains(portalItem.identifier) else {
```
This might make more sense.

In current version of the script, it doesn't write a k-v pair with `nil` value to the plist, so the change should work just fine. 

However, your condition reads like "if the portal item ID is in the plist, but the value is `nil`, then the item wasn't downloaded." Which doesn't read exactly right.

Reply:
I realized that I made wrong assumptions here: plist doesn't allow null values, as all the types that a plist supports aren't optional or NSNULL.